commandline/src/main/java/com/thoughtworks/go/util/command/CommandLine.java

```diff
@@ -319,6 +256,19 @@ public CommandLine withArgs(List<String> args) {
         return this;
     }
 
+    public CommandLine when(boolean condition, Consumer<CommandLine> thenDo) {
+        return this.tap((cmd) -> {
+            if (condition) {
+                thenDo.accept(cmd);
+            }
+        });
+    }
+
+    public CommandLine tap(Consumer<CommandLine> thenDo) {
+        thenDo.accept(this);
+        return this;
+    }
+
```

Comment:
This is the heart of this change (adding convenience methods); the rest is IntelliJ IDEA regrouping `private` and `protected` methods together at the bottom.

Reply:
@marques-work thanks for adding comments around the actual changes. Another option to consider is having a separate commit only for things like regrouping/refactoring.